File: medplum-java/medplum-model/src/main/java/com/fhir/FamilyMemberHistory.java
package com.fhir;

import com.StagedBuilderStyle;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import java.lang.Boolean;
import java.lang.String;
import java.util.List;
import java.util.Optional;
import org.immutables.value.Value;

@Value.Immutable
@StagedBuilderStyle
@JsonDeserialize(
    as = ImmutableFamilyMemberHistory.class
)
@JsonIgnoreProperties(
    ignoreUnknown = true
)
public interface FamilyMemberHistory {
  Optional<List<Reference>> reasonReference();

  Optional<Boolean> estimatedAge();

  Optional<Meta> meta();

  Optional<Age> deceasedAge();

  Optional<String> ageString();

  Optional<uri> implicitRules();

  Optional<List<Extension>> modifierExtension();

  Optional<Range> ageRange();

  Optional<CodeableConcept> dataAbsentReason();

  Optional<String> deceasedString();

  Optional<List<Annotation>> note();

  Optional<List<FamilyMemberHistory_Condition>> condition();

  Optional<String> bornDate();

  Optional<List<CodeableConcept>> reasonCode();

  Optional<String> deceasedDate();

  Optional<List<Identifier>> identifier();

  Optional<FamilymemberhistoryStatus> status();

  Optional<Period> bornPeriod();

  Optional<code> language();

  Optional<String> name();

  Optional<Boolean> deceasedBoolean();

  Optional<List<Extension>> extension();

  Optional<String> bornString();

  Optional<Age> ageAge();

  Optional<dateTime> date();

  Optional<List<ResourceList>> contained();

  Optional<List<canonical>> instantiatesCanonical();

  Optional<id> id();

  Optional<Narrative> text();

  Optional<Range> deceasedRange();

  Optional<CodeableConcept> sex();

  Optional<List<uri>> instantiatesUri();

  Reference patient();

  CodeableConcept relationship();

  String resourceType();

  static ImmutableFamilyMemberHistory.PatientBuildStage builder() {
    return ImmutableFamilyMemberHistory.builder();
  }
}

File: medplum-typescript/src/model/fhir/FamilyMemberHistory.ts
import { Age } from "./Age";
import { Annotation } from "./Annotation";
import { canonical } from "./canonical";
import { code } from "./code";
import { CodeableConcept } from "./CodeableConcept";
import { dateTime } from "./dateTime";
import { Extension } from "./Extension";
import { FamilymemberhistoryStatus } from "./FamilymemberhistoryStatus";
import { FamilyMemberHistory_Condition } from "./FamilyMemberHistory_Condition";
import { id } from "./id";
import { Identifier } from "./Identifier";
import { Meta } from "./Meta";
import { Narrative } from "./Narrative";
import { Period } from "./Period";
import { Range } from "./Range";
import { Reference } from "./Reference";
import { ResourceList } from "./ResourceList";
import { uri } from "./uri";

export interface FamilyMemberHistory {
    resourceType: string;
    id: id | null | undefined;
    meta: Meta | null | undefined;
    implicitRules: uri | null | undefined;
    language: code | null | undefined;
    text: Narrative | null | undefined;
    contained: ResourceList[] | null | undefined;
    extension: Extension[] | null | undefined;
    modifierExtension: Extension[] | null | undefined;
    identifier: Identifier[] | null | undefined;
    instantiatesCanonical: canonical[] | null | undefined;
    instantiatesUri: uri[] | null | undefined;
    status: FamilymemberhistoryStatus | null | undefined;
    dataAbsentReason: CodeableConcept | null | undefined;
    patient: Reference;
    date: dateTime | null | undefined;
    name: string | null | undefined;
    relationship: CodeableConcept;
    sex: CodeableConcept | null | undefined;
    bornPeriod: Period | null | undefined;
    bornDate: string | null | undefined;
    bornString: string | null | undefined;
    ageAge: Age | null | undefined;
    ageRange: Range | null | undefined;
    ageString: string | null | undefined;
    estimatedAge: boolean | null | undefined;
    deceasedBoolean: boolean | null | undefined;
    deceasedAge: Age | null | undefined;
    deceasedRange: Range | null | undefined;
    deceasedDate: string | null | undefined;
    deceasedString: string | null | undefined;
    reasonCode: CodeableConcept[] | null | undefined;
    reasonReference: Reference[] | null | undefined;
    note: Annotation[] | null | undefined;
    condition: FamilyMemberHistory_Condition[] | null | undefined;
}